src/Contract/Operation/Unfoldable.php

```diff
@@ -6,19 +6,23 @@
 
 use loophp\collection\Contract\Collection;
 
+/**
+ * @template TKey
+ * @template T
+ */
 interface Unfoldable
 {
     /**
      * Create a collection by yielding from a callback with an initial value.
      *
      * @see https://loophp-collection.readthedocs.io/en/stable/pages/api.html#unfold
      *
-     * @template T
+     * @template U
      *
-     * @param callable(T ...): list<T> $callback
-     * @param T ...$parameters
+     * @param callable((T|U) ...): list<T> $callback
+     * @param array<int, U> $parameters
```

Comment:
so does this mean that now in `$parameters` we want to be able to pass values of different types? What would be a good use case for that?

Reply:
Yes we could potentially do that.

Reply:
I gave a second look yesterday and you are perfectly right. I reverted some of the changes.

It's ready again for review.